internal/debug/flags.go

```diff
@@ -242,3 +254,28 @@ func Exit() {
 		closer.Close()
 	}
 }
+
+func getLogLocationURL(ctx *cli.Context) (*url.URL, error) {
+	locationURL, err := url.Parse(ctx.String(logLocationFlag.Name))
+	if err == nil {
+		if _, existErr := os.Stat(locationURL.Path); os.IsNotExist(existErr) {
+			// directory doesn't exist, create
+			createErr := os.Mkdir(locationURL.Path, os.ModePerm)
```

Comment:
Instead of checking then creating, you can just call `os.MkdirAll`. It creates if needed, leaves it be if it already exists. 
```
	if err := os.MkdirAll(locationURL.Path, os.ModePerm); err != nil {
		return nil, err
	}
```

Reply:
What I mean is, you can skip the `os.Stat` too